foundation/application.go

```diff
@@ -183,16 +186,32 @@ func setEnv() {
 		}
 	}
 
-	var env *string
-	if !flag.Parsed() && support.Env != support.EnvTest {
-		env = flag.String("env", ".env", "custom .env path")
-		flag.Parse()
-	} else {
-		testEnv := ".env"
-		env = &testEnv
+	env := getEnvPath()
+	if support.Env == support.EnvTest {
+		var (
+			relativePath string
+			envExist     bool
+			testEnv      = env
+		)
+
+		for i := 0; i < 50; i++ {
+			if _, err := os.Stat(testEnv); err == nil {
+				envExist = true
+
+				break
+			} else {
+				testEnv = filepath.Join("../", testEnv)
+				relativePath = filepath.Join("../", relativePath)
+			}
+		}
```

Comment:
Because the base path is the current folder when running a test case, the `.env` file needs to be found in the upper folder.